### PR TITLE
feat(es/lint): implement yoda rule

### PR DESCRIPTION
**Pull request by ArturAralin**
**Description:**

implement yoda rule

**BREAKING CHANGE:**
No

### CODE REVIEW

**Comment by kdy1** [comment]
This is also about code style, isn't it?
I'm not sure about this rule

**Comment by ArturAralin** [comment]
Yes. This rule restrict using literals on left side (e.g. 10 === a)

**Comment by alexander-akait** [comment]
@kdy1 Also you can always ping/assign me in an issue and I will give advice about rule/naming/options/etc

**Comment by kdy1** [comment]
Thanks! I'll review this soon.

File: crates/swc_ecma_lints/src/rules/yoda.rs [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    fn default() -> Self {
+        Self::Never
+    }
+}
+
+#[derive(Debug, Clone, Default, Copy, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct YodaConfig {
+    mode: Option<YodaConfigMode>,
+    except_range: Option<bool>,
```
Comment (kdy1) [inline crates/swc_ecma_lints/src/rules/yoda.rs:28]: I think we need `#[serde(default)]` for each fields.

File: crates/swc_ecma_lints/src/rules/yoda.rs [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+            if let (
+                Expr::Bin(BinExpr {
+                    op: op!("<") | op!("<="),
+                    ..
+                }),
+                Expr::Bin(BinExpr {
+                    op: op!(">") | op!(">="),
+                    ..
+                }),
+            ) = (bin_expr.left.as_ref(), bin_expr.right.as_ref())
```
Comment (kdy1) [inline crates/swc_ecma_lints/src/rules/yoda.rs:126]: I think this will skip checking of nested binary expression like `(foo === 1) && (1 === foo)`

**Review by kdy1 — CHANGES_REQUESTED** [review]
Thank you!

File: crates/swc_ecma_lints/src/rules/yoda.rs (hunk 2 above)
Comment (ArturAralin) [reply crates/swc_ecma_lints/src/rules/yoda.rs:126]: I was check it. It works correctly. I added your case to tests

**Review by kdy1 — APPROVED** [review]
Thank you!


---

swc-bump:
 - swc_ecma_lints

**Comment by kdy1** [comment]
CI is failing.
Can you run `UPDATE=1 cargo test` from `./crates/swc` after pulling the changes (merge commits for rebasing)

**Review by kdy1 — APPROVED** [review]

**Pull request merged**